File: cache/cache-annotation-processor/src/test/java/ru/tinkoff/kora/cache/annotation/processor/testcache/DummyCache22.java
package ru.tinkoff.kora.cache.annotation.processor.testcache;

import ru.tinkoff.kora.cache.annotation.Cache;
import ru.tinkoff.kora.cache.caffeine.CaffeineCache;
import ru.tinkoff.kora.cache.redis.RedisCache;

import java.math.BigDecimal;

@Cache("dummy22")
public interface DummyCache22 extends RedisCache<DummyCache22.Key, String> {

    record Key(String k1, BigDecimal k2) {}
}

File: cache/cache-symbol-processor/src/test/kotlin/ru/tinkoff/kora/cache/symbol/processor/testcache/DummyCache22.kt
package ru.tinkoff.kora.cache.symbol.processor.testcache

import ru.tinkoff.kora.cache.annotation.Cache
import ru.tinkoff.kora.cache.redis.RedisCache
import java.math.BigDecimal

@Cache("dummy22")
interface DummyCache22 : RedisCache<DummyCache22.Key, String> {

    data class Key(val k1: String, val k2: BigDecimal)
}
